<div class="container">

<table style="width: 100%;"><tr>
<td>read_genome</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Read genome files.</h2>

<h3>Description</h3>

<p>Genome files (UCSC "chromSize" files) contain chromosome name and size
information. These sizes are used by downstream functions to identify
computed intervals that have coordinates outside of the genome bounds.
</p>


<h3>Usage</h3>

<pre><code class="language-R">read_genome(path)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>path</code></td>
<td>
<p>containing chrom/contig names and sizes, one-pair-per-line,
tab-delimited</p>
</td>
</tr></table>
<h3>Value</h3>

<p>genome_df, sorted by <code>size</code>
</p>


<h3>Note</h3>

<p>URLs to genome files can also be used.
</p>


<h3>See Also</h3>

<p>Other read functions: 
<code>read_bed()</code>,
<code>read_vcf()</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">read_genome(valr_example("hg19.chrom.sizes.gz"))

## Not run: 
# `read_genome` accepts a URL
read_genome("https://genome.ucsc.edu/goldenpath/help/hg19.chrom.sizes")

## End(Not run)

</code></pre>


</div>