<div class="container">

<table style="width: 100%;"><tr>
<td>leukemia</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Expression levels recorded in leukemia patients.</h2>

<h3>Description</h3>

<p>Expression levels recorded for 3,571 genes in 72 patients with leukemia
(Golub et al, 1999). The binary outcome encodes the disease subtype:
acute lymphobastic leukemia (ALL) or acute myeloid leukemia (AML).
</p>


<h3>Usage</h3>

<pre><code class="language-R">data(leukemia)</code></pre>


<h3>Format</h3>

<p>Data are represented as a 72 x 3,571 matrix <code>x</code> of gene expression
values, and a vector <code>y</code> of 72 binary disease outcomes.
</p>


<h3>Source</h3>

<p>These are the preprocessed data of Dettling (2004) retrieved
from the supplementary materials accompanying Friedman et al (2010).</p>


<h3>References</h3>

<p>M. Dettling (2004). BagBoosting for tumor classification with gene
expression data. <em>Bioinformatics</em> <b>20</b>, 3583–3593.
</p>
<p>J. Friedman, T. Hastie and R. Tibshirani (2010). Regularization paths
for generalized linear models via coordinate descent. <em>Journal of 
Statistical Software</em> <b>33</b>, 1–22.
</p>
<p>T. R. Golub, et al. (1999). Molecular classification of cancer: class
discovery and class prediction by gene expression monitoring.
<em>Science</em> <b>286</b>, 531–537.
</p>


<h3>Examples</h3>

<pre><code class="language-R">  # See demo.leukemia.R vignette.
</code></pre>


</div>