<div class="container">

<table style="width: 100%;"><tr>
<td>merge.ides.annotation</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Merge potential iDES calls with variant annotation.</h2>

<h3>Description</h3>

<p>Merge potential iDES calls with variant annotation.
</p>


<h3>Usage</h3>

<pre><code class="language-R">## S3 method for class 'ides.annotation'
merge(ides.filename, output = TRUE,
  output.suffix = ".ann.txt",
  annovar.suffix.pattern = ".annovar.hg(\\d{2})_multianno.txt")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ides.filename</code></td>
<td>
<p>Path to formatted iDES output (typically from convert.ides.output file)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output</code></td>
<td>
<p>Logical indicating whether output should be saved to file. Defaults to true.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.suffix</code></td>
<td>
<p>Suffix to be appended to input filename if saving results to file</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annovar.suffix.pattern</code></td>
<td>
<p>Suffix to match ANNOAR file</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The VarDict variant calling includes a GATK call merging the call vcf file (allele frequency information etc.) with
the ANNOVAR annotation, and saving the result as a table. This function is an attempt to emulate that step 
for the iDES calls.
</p>


<h3>Value</h3>

<p>annotated.calls Data frame of annotations and iDES output.
</p>


</div>