<div class="container">

<table style="width: 100%;"><tr>
<td>run.varitas.pipeline.hybrid</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>run.varitas.pipeline.hybrid</h2>

<h3>Description</h3>

<p>Run VariTAS pipeline starting from both VCF files and BAM/ FASTQ files. 
Useful for processing data from the Ion PGM or MiniSeq where variant calling has been done on the machine, 
but you are interested in running more variant callers.
</p>


<h3>Usage</h3>

<pre><code class="language-R">run.varitas.pipeline.hybrid(vcf.specification, output.directory,
  run.name = NULL, fastq.specification = NULL,
  bam.specification = NULL, variant.callers = c("mutect", "vardict",
  "varscan", "lofreq", "muse"), proton = FALSE, quiet = FALSE,
  email = NULL, verify.options = !quiet,
  save.specification.files = !quiet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vcf.specification</code></td>
<td>
<p>Data frame containing details of vcf files to be processed. Must contain columns sample.id, vcf, and caller</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.directory</code></td>
<td>
<p>Main directory where all files should be saved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>run.name</code></td>
<td>
<p>Name of pipeline run. Will be added as a prefix to all LSF jobs.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fastq.specification</code></td>
<td>
<p>Data frame containing details of FASTQ files to be processed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>bam.specification</code></td>
<td>
<p>Data frame containing details of BAM files to be processed</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant.callers</code></td>
<td>
<p>Vector specifying which variant callers should be run.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proton</code></td>
<td>
<p>Logical indicating if data was generated by proton sequencing. Used to set base quality 
thresholds in variant calling steps.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>Logical indicating whether to print commands to screen rather than submit jobs. Defaults to FALSE,
can be useful to set to TRUE for testing.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>email</code></td>
<td>
<p>Email address that should be notified when pipeline finishes. If NULL or FALSE, no email is sent.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verify.options</code></td>
<td>
<p>Logical indicating whether to run verify.varitas.options</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>save.specification.files</code></td>
<td>
<p>Logical indicating if specification files should be saved to project directory</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class="language-R">run.varitas.pipeline.hybrid(
       bam.specification = data.frame(sample.id = c('Z', 'Y'), tumour.bam = c('Z.bam', 'Y.bam')),
       vcf.specification = data.frame(
         sample.id = c('a', 'b'),
         vcf = c('a.vcf', 'b.vcf'),
         caller = c('pgm', 'pgm')
       ),
       output.directory = '.',
       quiet = TRUE,
       run.name = "Test", 
       variant.callers = c('mutect', 'varscan')
     )

</code></pre>


</div>