<div class="container">

<table style="width: 100%;"><tr>
<td>run.variant.calling</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>run.variant.calling</h2>

<h3>Description</h3>

<p>Run variant calling for all samples
</p>


<h3>Usage</h3>

<pre><code class="language-R">run.variant.calling(bam.specification, output.directory,
  variant.callers = c("vardict", "mutect", "varscan", "lofreq", "muse"),
  paired = TRUE, proton = FALSE, sample.directories = TRUE,
  job.name.prefix = NULL, quiet = FALSE, verify.options = !quiet)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>bam.specification</code></td>
<td>
<p>Data frame containing details of BAM files to be processed, typically from <code>prepare.bam.specification</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>output.directory</code></td>
<td>
<p>Path to directory where output should be saved</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>variant.callers</code></td>
<td>
<p>Character vector of variant callers to be used</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>paired</code></td>
<td>
<p>Logical indicating whether to do variant calling with a matched normal</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>proton</code></td>
<td>
<p>Logical indicating whether data was generated by proton sequencing (ignored if running MuTect)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sample.directories</code></td>
<td>
<p>Logical indicating whether output for each sample should be put in its own directory (within output.directory)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>job.name.prefix</code></td>
<td>
<p>Prefix for job names on the cluster</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>quiet</code></td>
<td>
<p>Logical indicating whether to print commands to screen rather than submit the job</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verify.options</code></td>
<td>
<p>Logical indicating whether to run verify.varitas.options</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Run VarDict on each sample, and annotate the results with ANNOVAR. 
Files are output to a vardict/ subdirectory within each sample directory.
</p>


<h3>Value</h3>

<p>None
</p>


<h3>Examples</h3>

<pre><code class="language-R">run.variant.calling(
  data.frame(sample.id = c('Z', 'Y'), tumour.bam = c('Z.bam', 'Y.bam')),
  output.directory = '.',
  variant.caller = c('lofreq', 'mutect'),
  quiet = TRUE,
  paired = FALSE
)

</code></pre>


</div>