<div class="container">

<table style="width: 100%;"><tr>
<td>maf</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Minor allele frequency</h2>

<h3>Description</h3>

<p>Calculate the minor (or other) allele frequency.
</p>


<h3>Usage</h3>

<pre><code class="language-R">maf(x, element = 2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>an object of class vcfR or chromR</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>element</code></td>
<td>
<p>specify the allele number to return</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The function maf() calculates the counts and frequency for an allele.
A variant may contain more than two alleles.
Rare alleles may be true rare alleles or the result of genotyping error.
In an attempt to address these competing issues we sort the alleles by their frequency and the report statistics based on their position.
For example, setting element=1 would return information about the major (most common) allele.
Setting element=2 returns information about the second allele.
</p>


<h3>Value</h3>

<p>a matrix of four columns.
The first column is the total number of alleles, the second is the number of NA genotypes, the third is the count and fourth the frequency.
</p>


</div>