<div class="container">

<table style="width: 100%;"><tr>
<td>freq_peak_plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot freq_peak object</h2>

<h3>Description</h3>

<p>Converts allele balance data produced by <code>freq_peak()</code> to a copy number by assinging the allele balance data (frequencies) to its closest expected ratio.
</p>


<h3>Usage</h3>

<pre><code class="language-R">freq_peak_plot(
  pos,
  posUnits = "bp",
  ab1 = NULL,
  ab2 = NULL,
  fp1 = NULL,
  fp2 = NULL,
  mySamp = 1,
  col1 = "#A6CEE3",
  col2 = "#1F78B4",
  alpha = 44,
  main = NULL,
  mhist = TRUE,
  layout = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>pos</code></td>
<td>
<p>chromosomal position of variants</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>posUnits</code></td>
<td>
<p>units ('bp', 'Kbp', 'Mbp', 'Gbp') for 'pos' to be converted to in the main plot</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ab1</code></td>
<td>
<p>matrix of allele balances for allele 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ab2</code></td>
<td>
<p>matrix of allele balances for allele 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fp1</code></td>
<td>
<p>freq_peak object for allele 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>fp2</code></td>
<td>
<p>freq_peak object for allele 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mySamp</code></td>
<td>
<p>sample indicator</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col1</code></td>
<td>
<p>color 1</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col2</code></td>
<td>
<p>color 2</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>alpha</code></td>
<td>
<p>sets the transparency for dot plot (0-255)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>main</code></td>
<td>
<p>main plot title.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mhist</code></td>
<td>
<p>logical indicating to include a marginal histogram</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>layout</code></td>
<td>
<p>call layout</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>parameters passed on to other functions</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Creates a visualization of allele balance data consisting of a dot plot with position as the x-axis and frequency on the y-axis and an optional marginal histogram.
The only required information is a vector of chromosomal positions, however this is probably not going to create an interesting plot.
</p>


<h3>Value</h3>

<p>An invisible NULL.
</p>


<h3>See Also</h3>

<p>freq_peak,
peak_to_ploid
</p>


<h3>Examples</h3>

<pre><code class="language-R">
# An empty plot.
freq_peak_plot(pos=1:40)

data(vcfR_example)
gt &lt;- extract.gt(vcf)
hets &lt;- is_het(gt)
# Censor non-heterozygous positions.
is.na(vcf@gt[,-1][!hets]) &lt;- TRUE
# Extract allele depths.
ad &lt;- extract.gt(vcf, element = "AD")
ad1 &lt;- masplit(ad, record = 1)
ad2 &lt;- masplit(ad, record = 2)
freq1 &lt;- ad1/(ad1+ad2)
freq2 &lt;- ad2/(ad1+ad2)
myPeaks1 &lt;- freq_peak(freq1, getPOS(vcf))
is.na(myPeaks1$peaks[myPeaks1$counts &lt; 20]) &lt;- TRUE
myPeaks2 &lt;- freq_peak(freq2, getPOS(vcf), lhs = FALSE)
is.na(myPeaks2$peaks[myPeaks2$counts &lt; 20]) &lt;- TRUE
freq_peak_plot(pos = getPOS(vcf), ab1 = freq1, ab2 = freq2, fp1 = myPeaks1, fp2=myPeaks2)



</code></pre>


</div>