<div class="container">

<table style="width: 100%;"><tr>
<td>rePOS</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Create non-overlapping positions (POS) for VCF data</h2>

<h3>Description</h3>

<p>Converts allele balance data produced by <code>freq_peak()</code> to a copy number by assinging the allele balance data (frequencies) to its closest expected ratio.
</p>


<h3>Usage</h3>

<pre><code class="language-R">rePOS(x, lens, ret.lens = FALSE, buff = 0)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>x</code></td>
<td>
<p>a vcfR object</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>lens</code></td>
<td>
<p>a data.frame describing the reference</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ret.lens</code></td>
<td>
<p>logical specifying whether lens should be returned</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>buff</code></td>
<td>
<p>an integer indicating buffer length</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>Each chromosome in a genome typically begins with position one.
This creates a problem when plotting the data associated with each chromosome because the information will overlap.
This function uses the information in the data.frame <code>lens</code> to create a new coordinate system where chromosomes do not overlap.
</p>
<p>The data.frame <strong>lens</strong> should have a row for each chromosome and two columns.
The first column is the name of each chromosome as it appears in the vcfR object.
The second column is the length of each chromosome.
</p>
<p>The parameter <strong>buff</strong> indicates the length of a buffer to put in between each chromosome.
This buffer may help distinguish chromosomes from one another.
</p>
<p>In order to create the new coordinates the <code>lens</code> data.frame is updated with the new start positions.
The parameter 
</p>


<h3>Value</h3>

<p>Either a vector of integers that represent the new coordinate system or a list containing the vector of integers and the lens data.frame.
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Create some VCF data.
data(vcfR_example)
vcf1 &lt;-vcf[1:500,]
vcf2 &lt;-vcf[500:1500,]
vcf3 &lt;- vcf[1500:2533]
vcf1@fix[,'CHROM'] &lt;- 'chrom1'
vcf2@fix[,'CHROM'] &lt;- 'chrom2'
vcf3@fix[,'CHROM'] &lt;- 'chrom3'
vcf2@fix[,'POS'] &lt;- as.character(getPOS(vcf2) - 21900)
vcf3@fix[,'POS'] &lt;- as.character(getPOS(vcf3) - 67900)
vcf &lt;- rbind2(vcf1, vcf2)
vcf &lt;- rbind2(vcf, vcf3)
rm(vcf1, vcf2, vcf3)

# Create lens
lens &lt;- data.frame(matrix(nrow=3, ncol=2))
lens[1,1] &lt;- 'chrom1'
lens[2,1] &lt;- 'chrom2'
lens[3,1] &lt;- 'chrom3'
lens[1,2] &lt;- 22000
lens[2,2] &lt;- 47000
lens[3,2] &lt;- 32089

# Illustrate the issue.
dp &lt;- extract.info(vcf, element="DP", as.numeric=TRUE)
plot(getPOS(vcf), dp, col=as.factor(getCHROM(vcf)))

# Resolve the issue.
newPOS &lt;- rePOS(vcf, lens)
dp &lt;- extract.info(vcf, element="DP", as.numeric=TRUE)
plot(newPOS, dp, col=as.factor(getCHROM(vcf)))

# Illustrate the buffer
newPOS &lt;- rePOS(vcf, lens, buff=10000)
dp &lt;- extract.info(vcf, element="DP", as.numeric=TRUE)
plot(newPOS, dp, col=as.factor(getCHROM(vcf)))


</code></pre>


</div>