<div class="container">

<table style="width: 100%;"><tr>
<td>PR2.plot</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Parity rule 2 (PR2) plot</h2>

<h3>Description</h3>

<p>Make a Parity rule 2 (PR2) plot, where the AT-bias [A3/(A3 +T3)] at the third codon position of the four-codon amino acids of entire genes is the ordinate and the GC-bias [G3/(G3 +C3)] is the abscissa. The center of the plot, where both coordinates are 0.5, is where A = U and G = C (PR2), with no bias between the influence of the mutation and selection rates.
</p>


<h3>Usage</h3>

<pre><code class="language-R">PR2.plot(fasta.df)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>fasta.df</code></td>
<td>
<p>a data frame with seq_name and its DNA sequence.</p>
</td>
</tr></table>
<h3>Details</h3>

<p>For more information about PR2 plot <a href="https://www.tandfonline.com/doi/full/10.1038/emi.2016.106">Butt et al., 2016</a>.
</p>


<h3>Value</h3>

<p>A ggplot object.
</p>


<h3>Author(s)</h3>

<p>Ali Mostafa Anwar <a href="mailto:ali.mo.anwar@std.agr.cu.edu.eg">ali.mo.anwar@std.agr.cu.edu.eg</a> and Mohmed Soudy <a href="mailto:MohmedSoudy2009@gmail.com">MohmedSoudy2009@gmail.com</a>
</p>


<h3>Examples</h3>

<pre><code class="language-R">

# read DNA from fasta file
fasta &lt;- fasta.read("virus.fasta", "host.fasta")
fasta.v &lt;- fasta[[1]]
fasta.h &lt;- fasta[[2]]


PR2.plot(fasta.v)

</code></pre>


</div>