<div class="container">

<table style="width: 100%;"><tr>
<td>CUB</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Computes the Codon Usage Bias of DNA sequences
</h2>

<h3>Description</h3>

<p>The function reads aligned sequences in a fasta file and estimates the codon usage bias for each sequence. Several methods exist to estimate CUB; so far, only the "Effective Number of Codons" (ENC) calculation is available. 
</p>


<h3>Usage</h3>

<pre><code class="language-R">CUB(file = NULL, sequence = NULL, method = "ENC")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>file</code></td>
<td>
<p>FASTA file in which aligned sequences are stored.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>sequence</code></td>
<td>
<p>Alternatively, the result of seqinr::read.fasta.
</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>method</code></td>
<td>
<p>The method used to compute CUB. "ENC": Effective Number of Codons, as described in Wright (1990). 
</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A named vector of CUB scores. Names correspond to sequence names in the dataset. 
</p>


<h3>Author(s)</h3>

<p>Aurelie Hua-Van and Arnaud Le Rouzic.
</p>


<h3>References</h3>

<p>Wright, F. (1990). The 'effective number of codons' used in a gene. Gene, 87(1), 23-29.
</p>


<h3>See Also</h3>

<p><code>div</code>
</p>


<h3>Examples</h3>

<pre><code class="language-R">	seq.file &lt;- system.file("extdata/Genes", "Amd.fas", package="vhica")
	CUB(seq.file)
</code></pre>


</div>