<div class="container">

<table style="width: 100%;"><tr>
<td>prepare_map_custom_files</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Converts map information in custom format files to viewmap object</h2>

<h3>Description</h3>

<p>Converts map information in custom format files to viewmap object
</p>


<h3>Usage</h3>

<pre><code class="language-R">prepare_map_custom_files(dosages, phases, genetic_map, mks_pos = NULL)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>dosages</code></td>
<td>
<p>TSV or TSV.GZ file with both parents dosage information.
It should contain four columns: 1) character vector with chromosomes ID; 
2) Character vector with markers ID; 3) Character vector with parent ID; 
4) numerical vector with dosage.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>phases</code></td>
<td>
<p>TSV or TSV.GZ file with phases information. It should contain:
1) Character vector with chromosome ID; 2) Character vector with marker ID;
3 to (ploidy number)*2 columns with each parents haplotypes.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>genetic_map</code></td>
<td>
<p>TSV or TSV.GZ file with the genetic map information</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>mks_pos</code></td>
<td>
<p>TSV or TSV.GZ file with table with three columns: 1) marker ID; 
2) genome position; 3) chromosome</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>object of class <code>viewmap</code>
</p>


</div>