<div class="container">

<table style="width: 100%;"><tr>
<td>segreg_poly</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Polysomic segregation frequency - Function from MAPpoly</h2>

<h3>Description</h3>

<p>Computes the polysomic segregation frequency given a ploidy level
and the dosage of the locus in both parents. It does not consider
double reduction.
</p>


<h3>Usage</h3>

<pre><code class="language-R">segreg_poly(ploidy, d.p1, d.p2)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>ploidy</code></td>
<td>
<p>the ploidy level</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d.p1</code></td>
<td>
<p>the dosage in parent P</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>d.p2</code></td>
<td>
<p>the dosage in parent Q</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>a vector containing the expected segregation frequency for
all possible genotypic classes.
</p>


<h3>Author(s)</h3>

<p>Marcelo Mollinari, <a href="mailto:mmollin@ncsu.edu">mmollin@ncsu.edu</a>
</p>


<h3>References</h3>

<p>Mollinari, M., and Garcia, A.  A. F. (2019) Linkage
analysis and haplotype phasing in experimental autopolyploid
populations with high ploidy level using hidden Markov
models, _G3: Genes, Genomes, Genetics_. 
doi: <a href="https://doi.org/10.1534/g3.119.400378">10.1534/g3.119.400378</a>
</p>
<p>Serang O, Mollinari M, Garcia AAF (2012) Efficient Exact 
Maximum a Posteriori Computation for Bayesian SNP 
Genotyping in Polyploids. _PLoS ONE_ 7(2): 
e30906.
</p>


</div>