<div class="container">

<table style="width: 100%;"><tr>
<td>plot_map_list</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Plot a genetic map - Adapted from MAPpoly</h2>

<h3>Description</h3>

<p>This function plots a genetic linkage map(s)
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_map_list(viewmap, horiz = TRUE, col = "ggstyle", title = "Linkage group")
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>viewmap</code></td>
<td>
<p>object of class <code>viewmap</code></p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>horiz</code></td>
<td>
<p>logical. If FALSE, the maps are plotted vertically with the first map to the left. 
If TRUE  (default), the maps are plotted horizontally with the first at the bottom</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>col</code></td>
<td>
<p>a vector of colors for the bars or bar components (default = 'lightgrey')
<code>ggstyle</code> produces maps using the default <code>ggplot</code> color palette</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>title</code></td>
<td>
<p>a title (string) for the maps (default = 'Linkage group')</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A <code>data.frame</code> object containing the name of the markers and their genetic position
</p>


<h3>Author(s)</h3>

<p>Marcelo Mollinari, <a href="mailto:mmollin@ncsu.edu">mmollin@ncsu.edu</a>
</p>
<p>Cristiane Taniguti, <a href="mailto:chtaniguti@tamu.edu">chtaniguti@tamu.edu</a>
</p>


<h3>References</h3>

<p>Mollinari, M., and Garcia, A.  A. F. (2019) Linkage
analysis and haplotype phasing in experimental autopolyploid
populations with high ploidy level using hidden Markov
models, _G3: Genes, Genomes, Genetics_. 
doi: <a href="https://doi.org/10.1534/g3.119.400378">10.1534/g3.119.400378</a>
</p>


</div>