<div class="container">

<table style="width: 100%;"><tr>
<td>integrations</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Data on HIV integration sites from several studies</h2>

<h3>Description</h3>

<p>A dataset containing data from a meta-analysis looking for differences between active and inactive HIV integrations. Each row represents a provirus integrated somewhere in a human chromosome with whether viral expression was detectd, the distance to the nearest gene and the number of reads from H4K12ac ChIP-Seq mapped to within 50,000 bases of the integration.
</p>


<h3>Usage</h3>

<pre><code class="language-R">integrations
</code></pre>


<h3>Format</h3>

<p>A data frame with 12436 rows and 4 variables:
</p>

<dl>
<dt>study</dt>
<dd>
<p>the cell population infected by HIV</p>
</dd>
<dt>latent</dt>
<dd>
<p>whether the provirus was active (expressed) or inactive (latent)</p>
</dd>
<dt>nearestGene</dt>
<dd>
<p>distance to nearest gene (transcription unit) (0 if in a gene)</p>
</dd>
<dt>H4K12ac</dt>
<dd>
<p>number of reads aligned within +- 50,000 bases in a H4K12ac ChIP-Seq</p>
</dd>
</dl>
<h3>Source</h3>

<p><a href="https://retrovirology.biomedcentral.com/articles/10.1186/1742-4690-10-90">https://retrovirology.biomedcentral.com/articles/10.1186/1742-4690-10-90</a>, system.file("data-raw", "makeIntegrations.R", package = "vipor")
</p>


<h3>References</h3>

<p><a href="https://retrovirology.biomedcentral.com/articles/10.1186/1742-4690-10-90">https://retrovirology.biomedcentral.com/articles/10.1186/1742-4690-10-90</a>
</p>


</div>