<div class="container">

<table style="width: 100%;"><tr>
<td>snv_genome</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>snv_genome</h2>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data) and filtered (filter_variants) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class="language-R">snv_genome(vardf)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>vardf</code></td>
<td>
<p>A rearranged (arrange_data) and filtered (filtered_variants) vcf dataframe</p>
</td>
</tr></table>
<h3>Value</h3>

<p>A bar plot showing the number of variants per sample colored by their SNPEff annotation
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Example 1: Simple dataframe
df &lt;- data.frame(sample = c("m1", "m1", "m1", "m1", "m1",
                            "m2", "m2", "m2", "m2", "m2"),
			    annotation = c("downstrean_gene_variant", "synonymous_variant",
				                 "synonymous_variant", "stop_gained",
				                 "missense_variant", "downstrean_gene_variant",
				                 "downstrean_gene_variant", "synonymous_variant",
				                 "stop_gained", "missense_variant")
)

df

snv_genome(df)

# Example 2: Sample dataframe
snv_genome(example_filtered_SNV_df)

</code></pre>


</div>