<div class="container">

<table style="width: 100%;"><tr>
<td>arrange_data</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>arrange_data</h2>

<h3>Description</h3>

<p>Reads in a directory of VCF files and converts them into a single dataframe
</p>


<h3>Usage</h3>

<pre><code class="language-R">arrange_data(
  vardir,
  reference_fasta,
  annotated = "yes",
  ntlist = c("A", "G", "T", "C", "-"),
  verbose = FALSE
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>vardir</code></td>
<td>
<p>Directory path containing vcf files</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>reference_fasta</code></td>
<td>
<p>Reference fasta file used for alignment</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>annotated</code></td>
<td>
<p>Whether the VCF files are annotated using snpeff "yes" or "no" (default "yes")</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>ntlist</code></td>
<td>
<p>Nucleotides (default A, T, G, C) used for finding multiple alt alleles</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>verbose</code></td>
<td>
<p>set verbosity of the vcfR commands</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A large dataframe containing information from all input VCF files
</p>


</div>