<div class="container">

<table style="width: 100%;"><tr>
<td>plot_shannon</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>plot_shannon</h2>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data), filtered (filter_variants), and piped through the Shannon calculations (shannon_entropy) and outputs plots
</p>


<h3>Usage</h3>

<pre><code class="language-R">plot_shannon(shannon_df)
</code></pre>


<h3>Arguments</h3>

<table><tr style="vertical-align: top;">
<td><code>shannon_df</code></td>
<td>
<p>A dataframe that has been arranged (arrange_data), filtered (filter_variants), and piped through the Shannon calculations (shannon_entropy)</p>
</td>
</tr></table>
<h3>Details</h3>

<p>The 'plot_shannon()' function takes the variant dataframe and generates three plots.
1. The Shannon entropy, or amount of diversity, at each position in the genome at which a variant was found.
2. The Shannon entropy summed over each segment
3. The Shannon entropy summed over each genome
A higher value indicates more diversity.
</p>


<h3>Value</h3>

<p>Three plots showing the nt Shannon, chrom Shannon, and full genome Shannon calculations
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Sample dataframe
df &lt;- data.frame(sample = c("m1", "m2", "m1", "m2", "m1"),
                 CHROM = c("PB1", "PB1", "PB2", "PB2", "NP"),
                 POS = c(234, 234, 240, 240, 254),
                 minorfreq = c(0.010, 0.022, 0.043, 0.055, 0.011),
                 majorfreq = c(0.990, 0.978, 0.957, 0.945, 0.989),
                 SegmentSize = c(2280, 2280, 2274, 2274, 1809)
)

df

genome_size = 13133

# Modify the dataframe to add 5 new columns of shannon entropy data:
# 1. shannon_ntpos
# 2. chrom_shannon
# 3. genome_shannon
# 4. shannon_chrom_perkb
# 5. genome_shannon_perkb
shannon_df = shannon_entropy(df, genome_size)

# Plot
plot_shannon(shannon_df)

</code></pre>


</div>