<div class="container">

<table style="width: 100%;"><tr>
<td>filter_variants</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>filter_variants</h2>

<h3>Description</h3>

<p>Filters single-nucleotide variants using a coverage and frequency cutoff
</p>


<h3>Usage</h3>

<pre><code class="language-R">filter_variants(df, coverage_cutoff = 200, frequency_cutoff = 0.03)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>df</code></td>
<td>
<p>A rearranged VCF dataframe (rearranged using the arrange_data function)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>coverage_cutoff</code></td>
<td>
<p>The coverage cutoff for calling a SNV (default: 200x)</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>frequency_cutoff</code></td>
<td>
<p>Frequency cutoff for calling a SNV (default: 3%)</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A filtered VCF dataframe
</p>


<h3>Examples</h3>

<pre><code class="language-R">df &lt;- data.frame(CHROM = c("A", "B", "C"),
                POS = c(234, 240, 255),
                ALT_FREQ = c(0.016, 0.049, 0.031),
                gt_DP = c(716, 600, 187)
)

df

# Default: filter by 3% frequency threshold and 200 coverage cutoff
filter_variants(df)

# Example 1: A 1% allele frequency threshold and 200 coverage cutoff
filter_variants(df, coverage_cutoff = 200, frequency_cutoff = 0.01)

# Example 2: A 2% allele frequency threshold and 100 coverage cutoff
filter_variants(df, coverage_cutoff = 100, frequency_cutoff = 0.02)


</code></pre>


</div>