<div class="container">

<table style="width: 100%;"><tr>
<td>dNdS_segment</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>dNdS_segment</h2>

<h3>Description</h3>

<p>Reads in a dataframe that has been arranged (arrange_data), filtered (filter_variants), and annotated (prepare_annotations), calculates dNdS, and outputs plots
</p>


<h3>Usage</h3>

<pre><code class="language-R">dNdS_segment(annotation_df, SPLICEFORMS)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>annotation_df</code></td>
<td>
<p>A rearranged, filtered, and annotated vcf dataframe - must be for amino-acid specific calculations, cannot be the same as the dataframe used for SNP calculations</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>SPLICEFORMS</code></td>
<td>
<p>A character vector of isoform names</p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>A plot showing the dN/dS ratio for each splice form (rather than segment) for each sample
</p>


<h3>Examples</h3>

<pre><code class="language-R"># Sample Data
head(example_filtered_SNV_df)
dim(example_filtered_SNV_df)

# Plot showing the dN/dS ratio for each splice form
SPLICEFORMS = c("H1N1_PB2.1", "H1N1_PB1.1","H1N1_NS.2")
dNdS_segment(example_filtered_SNV_df, SPLICEFORMS)
</code></pre>


</div>