<div class="container">

<table style="width: 100%;"><tr>
<td>Error.Rate</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>
Error Rate</h2>

<h3>Description</h3>

<p>Calculates shared, singleton, and total second codon position error rate in a matrix of sequences.
</p>


<h3>Usage</h3>

<pre><code class="language-R">Error.Rate(single, shared, spec, seqlength)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>single</code></td>
<td>

<p>A vector of singleton very low frequency variant (VLF) counts for each position in the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>shared</code></td>
<td>

<p>A vector of shared very low frequenct variant (VLF) counts for each position in the sequence.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>spec</code></td>
<td>

<p>The number of specimen being considered in the dataset.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>seqlength</code></td>
<td>

<p>The length of the barcode sequence.</p>
</td>
</tr>
</table>
<h3>Details</h3>

<p>The arguments single and shared can be calculated simultaneously using the find.singles function.  The spec argument can be calculated by using the nrow() function on the sequence matrix.</p>


<h3>Value</h3>

<p>A vector containing the single, shared, and total error rate based on the number of second position VLFs.
</p>


<h3>Author(s)</h3>

<p>Taryn B. T. Athey and Paul D. McNicholas
</p>


<h3>Examples</h3>

<pre><code class="language-R">## Not run: data(birds)
species.names &lt;- birds[,2]
specimen.Number &lt;- nrow(birds)
rownames(birds) &lt;- species.names
Nuc.count &lt;- count.function(birds, specimen.Number, 648)
frequency.matrix &lt;- ffrequency.matrix.function(Nuc.count, 648)
birdSpec.freq &lt;- specimen.frequencies(frequency.matrix, birds, specimen.Number, species.names, 648)
Bird_specimen_VLFcount &lt;- VLF.count.spec(birdSpec.freq, 0.001, 648)
Bird_position_VLFcount &lt;- VLF.count.pos(birdSpec.freq, 0.001, 648)
bird_VLFconvert &lt;- VLF.convert.matrix(birds, birdSpec.freq, 0.001, 648)
bird_VLFnuc &lt;- VLF.nucleotides(bird_VLFconvert, birds, 648)
bird_VLFreduced &lt;- VLF.reduced(bird_VLFnuc, Bird_specimen_VLFcount, 648)
bird_species &lt;- separate(bird_VLFreduced)
birds_singleAndShared &lt;- find.singles(bird_species, 648)
Bird_error &lt;- Error.Rate(birds_singleAndShared[1,], birds_singleAndShared[2,], specimen.Number, 648)
## End(Not run)
</code></pre>


</div>