<div class="container">

<table style="width: 100%;"><tr>
<td>deseq_polar</td>
<td style="text-align: right;">R Documentation</td>
</tr></table>
<h2>Convert DESeq2 objects to a volcano3d object</h2>

<h3>Description</h3>

<p>This function is used instead of <code>polar_coords</code> if you have raw
RNA-Seq count data. It takes 2 <code>DESeqDataSet</code> objects, extracts statistical
results and converts the results to a 'volc3d' object, which can be directly
plotted.
</p>


<h3>Usage</h3>

<pre><code class="language-R">deseq_polar(
  object,
  objectLRT,
  contrast = NULL,
  data = NULL,
  pcutoff = 0.05,
  padj.method = "BH",
  filter_pairwise = TRUE,
  ...
)
</code></pre>


<h3>Arguments</h3>

<table>
<tr style="vertical-align: top;">
<td><code>object</code></td>
<td>
<p>An object of class 'DESeqDataSet' with the full design formula.
The function <code>DESeq</code> needs to have been run.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>objectLRT</code></td>
<td>
<p>An object of class 'DESeqDataSet' with the reduced design
formula. The function <code>DESeq</code> needs to have been run on this object with
argument <code>test="LRT"</code>.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>contrast</code></td>
<td>
<p>Character value specifying column within the metadata stored
in the DESeq2 dataset objects is the outcome variable. This column must
contain a factor with 3 levels. If not set, the function will select the
last term in the design formula of <code>object</code> as per DESeq2 convention.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>data</code></td>
<td>
<p>Optional matrix containing gene expression data. If not supplied,
the function will pull the expression data from within the DESeq2 object
using the DESeq2 function <code>assay()</code>. NOTE: for consistency with gene
expression datasets, genes are in rows.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>pcutoff</code></td>
<td>
<p>Cut-off for p-value significance</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>padj.method</code></td>
<td>
<p>Can be any method available in <code>p.adjust</code> or <code>"qvalue"</code>.
The option <code>"none"</code> is a pass-through.</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>filter_pairwise</code></td>
<td>
<p>Logical whether adjusted p-value pairwise statistical
tests are only conducted on genes which reach significant adjusted p-value
cut-off on the group likelihood ratio test</p>
</td>
</tr>
<tr style="vertical-align: top;">
<td><code>...</code></td>
<td>
<p>Optional arguments passed to <code>polar_coords</code></p>
</td>
</tr>
</table>
<h3>Value</h3>

<p>Calls <code>polar_coords</code> to return an S4 'volc3d' object
</p>


<h3>See Also</h3>

<p><code>polar_coords</code>, <code>voom_polar</code>,
<code>DESeq</code> in the DESeq2 package
</p>


<h3>Examples</h3>

<pre><code class="language-R">

  library(DESeq2)

  counts &lt;- matrix(rnbinom(n=1500, mu=100, size=1/0.5), ncol=15)
  cond &lt;- factor(rep(1:3, each=5), labels = c('A', 'B', 'C'))

  # object construction
  dds &lt;- DESeqDataSetFromMatrix(counts, DataFrame(cond), ~ cond)

  # standard analysis
  dds &lt;- DESeq(dds)

  # Likelihood ratio test
  ddsLRT &lt;- DESeq(dds, test="LRT", reduced= ~ 1)

  polar &lt;- deseq_polar(dds, ddsLRT, "cond")
  volcano3D(polar)
  radial_ggplot(polar)


</code></pre>


</div>